test.sh

```diff
@@ -160,12 +221,25 @@ test_coverage()
 
     echo "Testing Code Coverage"
     coverage erase
-    for testfile in tests/test_*.py
-    do
-        coverage run --append --source=. -m pytest -rsx -W ignore::RuntimeWarning -W ignore::DeprecationWarning -W ignore::UserWarning $testfile
-        check_errs $?
-    done
-    coverage report -m --fail-under=100 --skip-covered --omit=setup.py,docstring.py,min.py,stumpy/cache.py
+
+    # We always attempt to test everything but we may ignore things (ray, helper scripts) when we generate the coverage report
```

Comment:
It seems that `tests/test_ray.py` has not been included in the unit testing in Github Actions yet (or maybe I missed it?). Although the support for `ray` is experimental, I was wondering if that should be included in Github Actions. (And I think your goal was to do the same based on the phrase "always attempt to test everything..."

Reply:
No, I purposely kept `ray` testing separate from our testing due to the fact that `ray` development is slow and they don't even support Python 3.12 yet! `ray` testing would simply break our testing suite too often. Instead, I plan to move `ray` testing to our `stumpy_dev` `automation` so that it gets tested once per day instead

Reply:
> supported via pip for MacOS

Right! Noticed it when I was going through the ray documentation.

> Basically, test.sh will check whether ray is installed, if True then run the ray unit tests too but, if False, skip/ignore all ray tests.

Yes! And that was where I got confused as I noticed we do not install Ray in our GitHub Actions.  Now, I have the answer.